images/nested-libvirt/README.md

```diff
@@ -0,0 +1,50 @@
+# Nested libvirt on GCE
+
+This image enables launching a libvirt cluster nested in a GCE instance, for libvirt CI testing.
+
+This image contains [`nss_wrapper`](https://cwrap.org/nss_wrapper.html) to execute `ssh` commands as
+a mock user to interact with a GCE instance from an OpenShift container.
+
+OpenShift containers run with an arbitrary uid, but SSH requires a valid user.  `nss_wrapper`
+allows for the container's user ID to be mapped to a username inside of a container.
+
+### Example Usage
+
+You can override the container's current user ID and group ID by providing `NSS_WRAPPER_GROUP` and `NSS_WRAPPER_PASSWD` for the mock files, as well as `NSS_USERNAME`, `NSS_UID`, `NSS_GROUPNAME`, and/or `NSS_GID`. In OpenShift CI, `NSS_USERNAME` and `NSS_GROUPNAME` are set.  The random UID assigned to the container is the UID that the mock username is mapped to. 
+
+```sh
+podman run --rm -it \
+  -e NSS_WRAPPER_GROUP=/tmp/group \
+  -e NSS_WRAPPER_PASSWD=/tmp/passwd \
+  -e NSS_UID=1000 \
+  -e NSS_GID=1000 \
+  -e NSS_USERNAME=testuser \
+  -e NSS_GROUPNAME=testuser \
+  nss_wrapper_img sh -c 'mock-nss.sh; LD_PRELOAD=/usr/lib64/libnss_wrapper.so id testuser'
```

Comment:
Continuing from [here][1], here's using `mock-nss.sh` to `exec id`:

```console
$ git describe --dirty
v0.3.0-199-g8c3b446-dirty
$ git diff -U0
diff --git a/images/nested-libvirt/Dockerfile b/images/nested-libvirt/Dockerfile
index b99ef79..797eca4 100644
--- a/images/nested-libvirt/Dockerfile
+++ b/images/nested-libvirt/Dockerfile
@@ -3 +3 @@
-FROM openshift/origin-release:golang-1.10 AS build
+FROM docker.io/openshift/origin-release:golang-1.10 AS build
@@ -8 +8 @@ RUN hack/build.sh && hack/get-terraform.sh
-FROM centos:7
+FROM docker.io/library/centos:7
$ podman build -tag nss_wrapper_img -f images/nested-libvirt/Dockerfile .
$ podman run --rm -e NSS_WRAPPER_GROUP=/tmp/group   -e NSS_WRAPPER_PASSWD=/tmp/passwd   -e NSS_UID=1000   -e NSS_GID=1000   -e NSS_USERNAME=testuser   -e NSS_GROUPNAME=testuser nss_wrapper_img mock-nss.sh id testuser
uid=1000(testuser) gid=1000(testuser) groups=1000(testuser)
```

This works because of `mock-nss.sh`'s:

```console
$ git show origin/pr/625 | grep -B1 '^\+exec'
+export LD_PRELOAD=/usr/lib64/libnss_wrapper.so
+exec "$@"
```


[1]: https://github.com/openshift/installer/pull/625#discussion_r233263605

Reply:
yea, but that did not work on my end.  i'll look into it, agreed that `export LD_PRELOAD=/usr/lib64/libnss_wrapper.so` _shouldn't_ be required, so removing it